test/jasmine/tests/pie_test.js

```diff
@@ -522,33 +478,44 @@ describe('Test event property of interactions on a pie plot:', function() {
         it('should contain the correct fields', function() {
             mouseEvent('mouseover', pointPos[0], pointPos[1]);
 
-            var point0 = futureData.points[0],
-                evt = futureData.event;
-            expect(point0.originalEvent).toEqual(evt, 'points');
-            expect(evt.clientX).toEqual(pointPos[0], 'event.clientX');
-            expect(evt.clientY).toEqual(pointPos[1], 'event.clientY');
+            checkEventData(futureData);
+        });
+
+        it('should not emit a hover if you\'re dragging', function() {
+            gd._dragging = true;
+            mouseEvent('mouseover', pointPos[0], pointPos[1]);
+            expect(futureData).toBeUndefined();
+        });
+
+        it('should not emit a hover if hover is disabled', function() {
+            Plotly.relayout(gd, 'hovermode', false);
+            mouseEvent('mouseover', pointPos[0], pointPos[1]);
+            expect(futureData).toBeUndefined();
         });
     });
 
     describe('unhover events', function() {
         var futureData;
 
         beforeEach(function(done) {
-            Plotly.plot(gd, mockCopy.data, mockCopy.layout).then(done);
+            futureData = undefined;
+            Plotly.newPlot(gd, mockCopy.data, mockCopy.layout).then(done);
 
             gd.on('plotly_unhover', function(data) {
                 futureData = data;
             });
         });
 
         it('should contain the correct fields', function() {
+            mouseEvent('mouseover', pointPos[0], pointPos[1]);
             mouseEvent('mouseout', pointPos[0], pointPos[1]);
 
-            var point0 = futureData.points[0],
-                evt = futureData.event;
-            expect(point0.originalEvent).toEqual(evt, 'points');
-            expect(evt.clientX).toEqual(pointPos[0], 'event.clientX');
-            expect(evt.clientY).toEqual(pointPos[1], 'event.clientY');
+            checkEventData(futureData);
+        });
+
+        it('should not emit an unhover if you didn\'t first hover', function() {
+            mouseEvent('mouseout', pointPos[0], pointPos[1]);
+            expect(futureData).toBeUndefined();
```

Comment:
kind of a weird one... but lets say you dragged over the pie from another subplot, so there was no hover event - you shouldn't get an unhover event even if you mouse up and then mouse out of that slice.